db/schema.rb

```diff
@@ -264,6 +271,19 @@
     t.index ["service_id"], name: "index_notes_on_service_id"
   end
 
+  create_table "permissions", force: :cascade do |t|
```

Comment:
Is a Permission supposed to be the representation of an individual permission, or is it supposed to be a "many-to-many" association table? The name would suggest the former, but the fact that you have all the foreign keys on this table implies that it's the latter, where a row in this table can only represent at most one user and/or group at a time.

If this is just the latter, and if the plan isn't to have a DB object represent the former, then maybe we should give this a different name? In Django, I am used to Permissions being the word used to represent, for the lack of a better term, a "permission", the right to perform a certain action or access certain resources, and Permissions have a separate many-to-many join table with both Users and Groups. See https://blog.ovalerio.net/archives/1576 for an entity–relationship diagram.

I'm not sure if this is the best name, but perhaps we could call the many-to-many association something like an Entitlement, meaning whether a particular user or group is _entitled_ to some permission? I've seen the term used in similar contexts, such as in the following articles:

- https://arnon.dk/why-you-should-separate-your-billing-from-entitlement/
- https://developer.apple.com/documentation/bundleresources/entitlements

Reply:
I have updated this to follow the first scenario, that of a representation of an individual permission with a separate many-to-many join tables for both Users and Groups. Thanks for the detailed explanation! I think that following this system is more clear.